hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/NearCache.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2008-2013, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache.impl.nearcache;
+
+import com.hazelcast.config.InMemoryFormat;
+import com.hazelcast.monitor.NearCacheStats;
+
+/**
+ *
+ * @param <K> the type of the key
+ * @param <V> the type of the value
+ */
+public interface NearCache<K, V> {
```

Comment:
@mdogan Yes they are very similar, but since `com.hazelcast.client.nearcache.ClientNearCache` is used by other structures such as `IMap`, I didn't want to touch that part at this point. For example, JCache NearCache interface will also contact with new Eviction implementation and and `IMap` eviction is different so I din't want merge them into same hierarchy currently.